### PR TITLE
Contacts backup improvements/bugfixes

### PR DESCRIPTION
**Pull request by mario**
This PR will fix the following things once completely ready:

- refactor contacts backup into two separate fragments
- fix inability to scroll in the backup fragment
- fix fragment changes
- fix rotation by remembering the state
- fix contacts selection (currently broken)
- fix several bugs that cause the application to crash
- various other issues

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
@mario I think the package names should be all lower case, right?

**Comment by mario** [comment]
Sure, I fixed that already

File: src/main/res/values/strings.xml [hunk 1]
```diff
@@ -634,6 +634,7 @@
     <string name="prefs_category_about">About</string>
 
     <string name="actionbar_contacts">Contacts backup</string>
+    <string name="actionbar_contacts_restore">Contacts restore</string>
```
Comment (AndyScherzinger) [inline src/main/res/values/strings.xml]: @mario @tobiasKaminsky just a question to discuss, shouldn't we call it "Backup contacts" and "Restore contacts"? Seems more natural, grammatically correct to me. :)

**Comment by AndyScherzinger** [comment]
👍 looks good code wise. Just one minor issue about the titles to be discussed

File: src/main/res/values/strings.xml (hunk 1 above)
Comment (mario) [inline src/main/res/values/strings.xml]: I can change this, sure.

**Comment by mario** [comment]
Thanks, but you'll have to do it again once this PR is ready @AndyScherzinger :p

**Comment by mario** [comment]
@AndyScherzinger @tobiasKaminsky done. Approve, please. Thanks.

**Comment by mario** [comment]
@AndyScherzinger go approve :P

**Comment by AndyScherzinger** [comment]
:+1: ups, forget that one. Reviewed code and tested successfully

**Comment by mario** [comment]
@tobiasKaminsky please don't remember to approve this :P

**Comment by mario** [comment]
@AndyScherzinger can you rebase this pls?

**Comment by mario** [comment]
@tobiasKaminsky please review & merge.

**Comment by tobiasKaminsky** [comment]
- no restore button?
- enabling contacts backup via toggle (and no permission) asks for permission, but then do not switch the toggle (was working before)
- clicking on a vcf file showed the contact list (not working anymore)

**Comment by mario** [comment]
So comments...

1) Restore button appears once you select some contacts to restore
2) I haven't touched the toggle either, but I will take a look
3) Clicking on a VCF file where? I don't think I touched that code...
4) Was the bottom bar there before? I don't remember removing it, but can add it.

**Comment by AndyScherzinger** [comment]
👍 code wise fine, fixed some smaller issues directly, like printStackTrace and rebased against master (so the bottom bar is back).

**Comment by tobiasKaminsky** [comment]
I meant the restore / choose date thing. 
It is not shown right after a re-install, even if contacts-backup are downloaded.

**Comment by mario** [comment]
OK so @AndyScherzinger fixed bottom bar, I fixed the restore button. I'll have to refactor the vcf file click because it's strange, and I'll just launch an activity rather than opening a fragment to the right.

Looking at toggle now.

**Comment by mario** [comment]
@tobiasKaminsky all issues should be fixed now.

**Comment by mario** [comment]
Somebody should maybe check if automatic backups work with my changes - I forgot to test that part :)

**Comment by mario** [comment]
@tobiasKaminsky please fix what Andy mentioned and approve. :P

**Comment by mario** [comment]
OK, go go go :D

Both of you, approve! :)

**Comment by tobiasKaminsky** [comment]
- enable "contacts backup" -> toggle switches to on
- deny contact permission
- toggle stays turned on

- when in backup contacts view, open drawer: should highlight "backup contact" (in regular build)

**Comment by tobiasKaminsky** [comment]
Ups :/ 
I have pushed the fix for #885 to this branch.
Shall I revert @mario?

**Comment by AndyScherzinger** [comment]
Also related to contacts, so I'd say let push this one to DONE :)

**Comment by mario** [comment]
@AndyScherzinger I have to fix the issues @tobiasKaminsky mentioned + review this change @tobiasKaminsky just did - but I'm fine having it in this branch, so no need for revert.

**Comment by mario** [comment]
@AndyScherzinger @tobiasKaminsky I have fixed the two bugs. However, while doing the inspection of the code @tobiasKaminsky changes, I've noticed the following (missed that before):

- the automatic backup toggle is stored in preferences globally
- last backup time is stored globally

This produces the following issues:
- wrong backup time is shown if you have multiple accounts
- automatic backup disabled for all accounts if you disable it for one of them

@tobiasKaminsky please fix.

**Comment by mario** [comment]
Done. @AndyScherzinger can you approve pls?

**Comment by tobiasKaminsky** [comment]
To clarify for @AndyScherzinger: I fixed those issues that @Mario found here https://github.com/nextcloud/android/pull/874#issuecomment-297912530

File: src/main/java/com/owncloud/android/datamodel/ArbitraryDataProvider.java [hunk 2]
```diff
@@ -0,0 +1,233 @@
+/**
+ * Nextcloud Android client application
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/datamodel/ArbitraryDataProvider.java]: Please remove the paragraph tags :)

**Comment by AndyScherzinger** [comment]
👍 Fine with me, only found the paragraph tags issue plus of course codacy's findings about unused code that should be fixed too :)

File: src/main/java/com/owncloud/android/datamodel/ArbitraryDataProvider.java (hunk 2 above)
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/datamodel/ArbitraryDataProvider.java]: This is done by AndroidStudio's code formatter...
If we want to use the defaults we have to keep it in, or every contributor has to customize this.
(or we find a way to include the code formatter style in our project and use it automatically)

**Pull request merged**